### ISSUE TITLE
BART : how to reproduce CNN/DM results ?

### ISSUE THREAD

**Issue by Colanim** [pytorch/fairseq#1364]
**How can I fine-tune BART on the CNN/DM dataset ?** 

Also, are you going to release fine-tuned weights on CNN/DM dataset, like for MNLI ?

**Comment by magic282** [comment]
Same question here. How can we finetune a seq2seq tasks using BART?

**Comment by ngoyal2707** [comment]
@Colanim Yes, we plan to release CNN finetuned model soon.

**Comment by memray** [comment]
@ngoyal2707 Can you also clarify the details for fine-tuning, say the optimizer, learning rate, warm-up step etc.? Thank you!

**Comment by ngoyal2707** [comment]
@yinhanliu can you please add above details?

**Comment by yinhanliu** [comment]
@memray during ft on cnn/dm, we use detokenized cnn/dm data, and process it through gpt2. the best model was obtained by 3e-05 lr, steps 20k with batch size 80 sentences, we used Adam for optimizer and lr linearly decayed lr scheduler. Also, the input to encoder is document with truncation of 1024 max position. During decoding, min Length of 55, max length of 140, beam size of 4 and length penalty of 2.0 were used to achieve the best result.

**Comment by memray** [comment]
@yinhanliu Awesome! Thank you very much for the information and I'll try it out. BTW I noticed that the [results of CNNDM](https://github.com/pytorch/fairseq/blob/6f7b7d202199dc1534a77a247da1ad4604c21baa/examples/bart/README.md) are on dev set. Does it mean validation set instead of test set?

Thanks again,
Rui

**Comment by yinhanliu** [comment]
@memray the table in readme result is on dev. test result is in paper.

**Comment by memray** [comment]
@yinhanliu Two results are the same. Maybe one should be updated?

**Comment by Colanim** [comment]
@ngoyal2707 @yinhanliu Thanks for releasing the fine-tuned model on CNN/DM as well as the test procedure !! ❤️ 

May I ask if you plan to also open-source the training procedure ? So we can also fine-tune our own models ?

**Comment by yinhanliu** [comment]
@memray  thanks for pointing this out. we checked the numbers, and it was a typo. everything we showed here is on test.

**Comment by yinhanliu** [comment]
@Colanim yes. shortly. we just released the evaluate code and model.

**Comment by magic282** [comment]
Thanks for releasing the fine-tuned model on CNN/DM. Will the fine-tune code be released as the GLUE tasks?

**Comment by Colanim** [comment]
@magic282 https://github.com/pytorch/fairseq/blob/master/examples/bart/README.glue.md

**Comment by magic282** [comment]
@Colanim Sorry, I meant the seq2seq tasks. Is the fine-tuning procedure same as GLUE?

**Comment by ngoyal2707** [comment]
@magic282 Yes, we will release those in sometime

**Comment by saverymax** [comment]
Hi there, it doesn't appear that the code for fine-tuning BART on CNN/DM was ever released. Are you able to publish that here? Or could you potentially share a gist of it? Thank you!

**Comment by saverymax** [comment]
Hi there, I cannot find any script for fine-tuning BART on CNN/DM. Was this ever released? Thank you!

**Comment by 311dada** [comment]
> @memray during ft on cnn/dm, we use detokenized cnn/dm data, and process it through gpt2. the best model was obtained by 3e-05 lr, steps 20k with batch size 80 sentences, we used Adam for optimizer and lr linearly decayed lr scheduler. Also, the input to encoder is document with truncation of 1024 max position. During decoding, min Length of 55, max length of 140, beam size of 4 and length penalty of 2.0 were used to achieve the best result.

Could you please clarify similar details for XSUM? Thank you!
